src/traces/scatter/style.js

```diff
@@ -24,9 +24,16 @@ module.exports = function style(gd) {
 
     s.selectAll('g.points')
         .each(function(d) {
-            d3.select(this).selectAll('path.point')
-                .call(Drawing.pointStyle, d.trace || d[0].trace);
-            d3.select(this).selectAll('text')
+            var el = d3.select(this);
+            var pt = el.selectAll('path.point');
+
+            pt.call(Drawing.pointStyle, d.trace || d[0].trace);
+
+            pt.each(function(cd) {
+                d3.select(this).classed('plotly-customdata', cd.data !== null && cd.data !== undefined);
```

Comment:
Just a little style point: `pt` makes it sounds like it refers to a single point, but it's all points so I'd change it to `pts`. And elsewhere `cd` refers to a whole `calcdata` array so this tripped me up for a sec - maybe change it to `pointData` or something.